### PR TITLE
support china

### PR DESCRIPTION
**Pull request by nirLumigo**

### CODE REVIEW

File: src/lumigo_log_shipper/lumigo_shipper.py [hunk 1]
```diff
@@ -25,30 +23,14 @@ def ship_logs(aws_event: dict, programmatic_error_keyword: str = None) -> int:
...
         return res
+    except ChinaMissingEnvVar:
+        pass
     except Exception as e:
         # lumigo_shipper will print out the exception but won't raises it
         get_logger().critical("Failed to send customer logs", exc_info=e)
     return 0
 
 
-def _ship_aws_logs(aws_logs: List[Dict], programmatic_error_keyword: str = None) -> int:
```
Comment (nirLumigo) [inline src/lumigo_log_shipper/lumigo_shipper.py:34]: This function isn't used anywhere, so I deleted it and the relevant tests

File: src/lumigo_log_shipper/utils/aws_utils.py [hunk 2]
```diff
@@ -12,14 +13,19 @@ def extract_aws_logs_data(event: dict) -> AwsLogSubscriptionEvent:
...
     logs_data_dict = json.loads(logs_data_unzipped)
+    logs_data_dict["region"] = get_current_region()
     return dacite.from_dict(AwsLogSubscriptionEvent, logs_data_dict)
 
 
-def extract_aws_logs_data_from_log_record(
-    logs_data_dict: dict,
-) -> AwsLogSubscriptionEvent:
-    return dacite.from_dict(AwsLogSubscriptionEvent, logs_data_dict)
+def am_i_in_china() -> bool:
```
Comment (saartochner) [inline src/lumigo_log_shipper/utils/aws_utils.py:20]:
```suggestion
def is_china_region() -> bool:
```
Is better IMO 😂

File: src/lumigo_log_shipper/utils/aws_utils.py [hunk 3]
```diff
@@ -12,14 +13,19 @@ def extract_aws_logs_data(event: dict) -> AwsLogSubscriptionEvent:
...
+def get_current_region() -> Optional[str]:
+    return os.environ.get("AWS_REGION")
 
 
-def get_current_region() -> str:
-    return os.environ.get("AWS_REGION", "us-west-2")
+def get_dest_region() -> str:
+    if am_i_in_china():
+        return "us-west-2"
+    return get_current_region() or "us-west-2"
```
Comment (saartochner) [inline src/lumigo_log_shipper/utils/aws_utils.py:31]: extract to cost DEFAULT_REGION

File: src/lumigo_log_shipper/utils/sts.py [hunk 4]
```diff
@@ -1,11 +1,42 @@
...
+        f"Failed to send customer logs because {env_var_name} env var is missing and it is needed in China"
+    )
+    raise ChinaMissingEnvVar()
+
+
+def _get_boto_sts_client():
+    if not am_i_in_china():
+        return boto3.client("sts")
+    aws_access_key_id = _get_china_env_var("LUMIGO_LOGS_EDGE_AWS_ACCESS_KEY_ID")
+    aws_secret_access_key = _get_china_env_var("LUMIGO_LOGS_EDGE_AWS_SECRET_ACCESS_KEY")
```
Comment (saartochner) [inline src/lumigo_log_shipper/utils/sts.py:22]: This is a customer's lambda.
Does he have these secrets?

File: src/test/utils/test_aws_utils.py [hunk 5]
```diff
@@ -27,47 +28,24 @@ def test_extract_aws_logs_data_simple_flow(simple_aws_event):
...
+
+
+def test_get_dest_region(monkeypatch):
+    monkeypatch.setenv("AWS_REGION", "us-east-1")
+    assert get_dest_region() == "us-east-1"
+
+
+def test_get_dest_region_china(monkeypatch):
+    monkeypatch.setenv("AWS_REGION", "cn-northwest-1")
+    assert get_dest_region() == "us-west-2"
```
Comment (saartochner) [inline src/test/utils/test_aws_utils.py:51]: I would have used parametrized

**Review by saartochner — APPROVED** [review]
great work!

File: src/lumigo_log_shipper/utils/sts.py (hunk 4 above)
Comment (nirLumigo) [reply src/lumigo_log_shipper/utils/sts.py:22]: We'll need to give him the credentials and tell him to add these env vars.
We do the same for sending traces in China.

**Pull request merged**